src/v2/models/AppState.ts

```diff
@@ -252,17 +254,31 @@ export default class AppState extends Model {
     this.trigger('cache:clear');
   }
 
+  chooseRemediation(transformedResponse): IdxRemediation | undefined {
+    if (_.isEmpty(transformedResponse.remediations)) {
+      return;
+    }
+
+    // Special case: show select enrollment channel instead of QR code on mobile
+    if ((BrowserFeatures.isAndroid() || BrowserFeatures.isIOS()) && transformedResponse.currentAuthenticator?.contextualData?.selectedChannel === 'qrcode') {
+      return transformedResponse.remediations.find(r => r.name === 'select-enrollment-channel')
+    }
```

Comment:
- This is specific to the OV enroll, why move up to here?
- can it be moved to https://github.com/okta/okta-signin-widget/blob/master/src/v2/models/AppState.ts#L138 if we have to?

Reply:
Technically this refactor makes broader impact.
- It used to say `during the OV enroll && is Android or IOS &&  selectedChannel is qrcode, render select enrollment channel form`
- now it says `is Android or IOS &&  selectedChannel is qrcode, render select enrollment channel form`


Unless `contextualData?.selectedChannel === 'qrcode'` is isomorphism to `is OV enroll flow`, we may screw up other flows?

Reply:
maybe hold off the refactor if possible to minify impart?

Reply:
I could add an additional check that it also matches these constraints, which was done by ViewFactory:
```
  [RemediationForms.ENROLL_POLL]: {
    [AUTHENTICATOR_KEY.OV]: EnrollPollOktaVerifyView,
  },
 ```
 The first remediation shold be `enroll-poll` and the authenticator key should be "okta_verify"

Reply:
This refactor is going in next major version, it will be tested extensively before release, which has no specific deadline.